core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/kit/KitComponentImpl.java

```diff
@@ -0,0 +1,229 @@
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+package uk.ac.bbsrc.tgac.miso.core.data.impl.kit;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Transient;
+
+import org.joda.time.LocalDate;
+import org.springframework.format.annotation.DateTimeFormat;
+
+import com.eaglegenomics.simlims.core.Note;
+
+import uk.ac.bbsrc.tgac.miso.core.data.KitComponent;
+import uk.ac.bbsrc.tgac.miso.core.data.KitComponentDescriptor;
+import uk.ac.bbsrc.tgac.miso.core.util.DateUtils;
+
+/**
+ * Skeleton implementation of a KitComponent
+ *
+ * @author Rob Davey, Michal Zak
+ * @since 0.0.2
+ */
+public class KitComponentImpl implements KitComponent {
+  public static final Long UNSAVED_ID = 0L;
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long kitComponentId = KitComponentImpl.UNSAVED_ID;
+  private String identificationBarcode;
+  private String locationBarcode;
+
+  @Transient
+  @Enumerated(EnumType.STRING)
+  private Collection<Note> notes = new HashSet<>();
+  private String lotNumber;
+  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+  private LocalDate kitReceivedDate;
+  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+  private LocalDate kitExpiryDate;
+  private boolean exhausted;
+
+  private KitComponentDescriptor kitComponentDescriptor;
+
+  @Override
+  @Deprecated
+  public Long getKitComponentId() {
+    return kitComponentId;
+  }
+
+  @Override
+  @Deprecated
+  public void setKitComponentId(Long kitComponentId) {
+    this.kitComponentId = kitComponentId;
+  }
+
+  @Override
+  public long getId() {
+    return kitComponentId;
+  }
+
+  @Override
+  public void setId(long id) {
+    this.kitComponentId = id;
+  }
+
+  // TODO: NOT SURE ABOUT THIS ONE
+  @Override
+  public String getName() {
+    // it should be KitDescriptor.name + kitComponentDescriptor.name
+    return getKitComponentDescriptor().getName();
+  }
+
+  @Override
+  public String getLotNumber() {
+    return lotNumber;

```

Comment:
Does the lot number apply to a single bottle of stuff, or does it apply to all kitComponents that arrive in a single kit box?